### PR TITLE
[Move Expando]LPS-191477 Cleanup, remove unused methods from PortalUtil, Portal and…

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPS-191477

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 1]
```diff
@@ -2600,42 +2449,6 @@ else if (type == ExpandoColumnConstants.STRING_LOCALIZED) {
 		return value;
 	}
 
-	@Override
-	public Serializable getExpandoValue(
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:2604]: @slnn It seems you are removing a lot of methods, are they related to Expando, or you just want a complete clean-up?

**Comment by catarineaguiar** [comment]
no regressions found

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 1 above)
Comment (slnn) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:2604]: Actually, Expando just needs to remove part of them, but I want a complete clean-up.

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 2]
```diff
@@ -2880,7 +2693,7 @@ public String getJsSafePortletId(String portletId) {
 
 	@Override
 	public String getLayoutActualURL(Layout layout) {
-		return getLayoutActualURL(layout, getPathMain());
+		return _getLayoutActualURL(layout, getPathMain());
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:2696]: @slnn You replace these calls with a private method, but I don't see where your logic comes from. Did you also plan to remove `public String getLayoutActualURL(Layout layout, String mainPath)`?

Comment (slnn) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:2696]: Oh, my bad, I planned to remove public String getLayoutActualURL(Layout layout, String mainPath), but I found it is not unused, however, I forget to remove the temporary change,  I will double check and force push.

Comment (slnn) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:2696]: Oh, my bad, I planned to remove public String getLayoutActualURL(Layout layout, String mainPath), but I found it is not unused, however, I forget to remove the temporary change,  I will double check and force push.

**Comment by slnn** [comment]
ci:test:sf

**Comment by slnn** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 1 above)
Comment (slnn) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:2604]: I have updated and retriggered ci tests

**Comment by Kvale1733** [comment]
@slnn No regressions found.

**Review by dantewang — APPROVED** [review]

**Comment by tinatian** [comment]
ci:test:licensing

**Comment by tinatian** [comment]
ci:test:license

**Comment by dantewang** [comment]
@tinatian This passed license tests.

**Comment by tinatian** [comment]
@slnn Please do:
1. Remove expando related api only in this pull
2. Create a new ticket to remove unused methods in Portal interface. I would like to know the detailed reason why it is safe to remove the API. For example, the api was added by a feature and now the feature is removed and no one is using it. It is risky to me to remove so many interface methods together in one pull without any reason, even though we are not using those api methods, it may still break customer's code.

Please resend.

Thanks.
Tina.

**Comment by slnn** [comment]
> @slnn Please do:
> 
> 1. Remove expando related api only in this pull
> 2. Create a new ticket to remove unused methods in Portal interface. I would like to know the detailed reason why it is safe to remove the API. For example, the api was added by a feature and now the feature is removed and no one is using it. It is risky to me to remove so many interface methods together in one pull without any reason, even though we are not using those api methods, it may still break customer's code.
> 
> Please resend.
> 
> Thanks. Tina.

Remove expando related api only in https://github.com/liferay-core-infra/liferay-portal/pull/5181

The removal of other unused methods in Portal interface is on the way.

**Pull request closed without merging**